### ISSUE TITLE
建议官方整理  TagUI 的库文档

### ISSUE THREAD

**Issue by kangyiwen** [aisingapore/TagUI#1296]
@ruthtxh  @kensoh 很多用户在使用 TagUI 的过程中，会遇到各种各样的功能疑问，这个功能有没有，那个功能有没有。
TagUI 支持 JS 和 PY，功能肯定都能实现，但不知道怎么用。

其中有几个用户，放弃了TagUI 去学 Robocorp ，理由很简单，因为有更加完备的文档   [https://rpaframework.org/](url)

相比之下，TagUI 的文档简单了点，希望 TagUI 可以出一个类似 Robocorp 更加完整的文档，把功能和语法都列出来。
我认为，更加完备的文档，可能比功能迭代更重要。

**Comment by ruthtxh** [comment]
Hi @kangyiwen thanks for the feedback, do you have some examples of steps that requires more detailed documentation? So we can understand what is lacking from the current documentation.

**Comment by kangyiwen** [comment]
例如 大家会需要 TagUI 对 Excel 的操作代码例子，比如下面这些对Excel的操作，对应到 TagUI 应该怎么办？
![image](https://user-images.githubusercontent.com/76672577/199287407-887df345-5279-422c-b051-a107b2c27e43.png)

**Comment by kangyiwen** [comment]
用户其实非常“懒”，TagUI 可以用  PY 或 JS 来处理Excel的动作，但大家就是希望有一个官方提供的详细文档，照抄就行；而不是再去各处搜索 PY 如何处理 Excel。

**Comment by ruthtxh** [comment]
Hi @kangyiwen not all the steps you see in this list can be applied to how TagUI's excel step works. For example, to write to excel, for TagUI we will use the [write syntax](https://tagui.readthedocs.io/en/latest/reference.html?highlight=excel#write) [workbook]sheet!cell = variable which will automatically create a new file and create a new sheet if it is not already there. However from what I see in the list you sent, some possible steps associated with writing to excel might be Create Workbook > Create Worksheet > Set Cell Value > Save Workbook > Close Workbook. 

For TagUI most of the time multiple actions are carried out automatically in one step hence the documentation is more concise.

**Comment by kensoh** [comment]
Also adding on, admittedly (I created the Excel integration) the TagUI Excel integration does not cover the large breadth of actions doable with Excel, but the most commonly used ones. The main reason is both to simplify API and also huge resources is needed to create Excel integrations. I coded the logic to integrate directly on Windows and Mac Excel without any dependency. Not sure if Robocorp Excel integration can work on Mac or not.

But anyway, I'll agree that TagUI various areas might not be as good as Robocorp due to the deep VC funding and their team size (40+?). So that is the pros and cons of using a truly open-source project without any paid features / subscriptions. Robocorp may seem well-featured but like other commercialised open-source tools, without paying any money, a developer might not get far.
